java/ql/src/experimental/Security/CWE/CWE-094/SpringViewManipulation.ql

```diff
@@ -13,7 +13,15 @@ import java
 import SpringViewManipulationLib
 import DataFlow::PathGraph
 
-from DataFlow::PathNode source, DataFlow::PathNode sink, SpringViewManipulationConfig conf
-where conf.hasFlowPath(source, sink)
+from
+  DataFlow::PathNode source, DataFlow::PathNode sink, SpringViewManipulationConfig conf,
+  StringFlowConfig s
+where
+  thymeleafIsUsed() and
+  conf.hasFlowPath(source, sink) and
+  not s.hasFlow(_, sink.getNode()) and
+  not exists(AddExpr e | e.getLeftOperand().(StringLiteral).getRepresentedString() = "redirect:" |
```

Comment:
IIRC there are other kind of redirection prefixes that can be used such as `ajaxredirect:`